src/index/import/markdown.ts

```diff
@@ -145,11 +147,24 @@ export function markdownImport(
             list.id,
             list.task
         );
+        let content = lines.slice(item.start, item.end + 1).join("\n");
+        /** strip inline fields maybe */
+
+        let marker = content.split("\n")[0].replace(markerRegex, "").trim().slice(0, 1);
+        item.text = content.replace(contentRegex, "");
+        item.symbol = marker;
         listItems.set(item.start, item);
     }
 
     // In the second list pass, actually construct the list heirarchy.
     for (const item of listItems.values()) {
+        let content = lines.slice(item.start, item.end + 1).join("\n");
```

Comment:
Do you need to do this logic twice? This second loop is not actually loading new list items, it's just setting up the tree structure.